internal/datanode/flow_graph_insert_buffer_node.go

```diff
@@ -549,16 +533,14 @@ func (ibNode *insertBufferNode) bufferInsertMsg(msg *msgstream.InsertMsg, endPos
 
 	// update buffer size
 	buffer.updateSize(int64(msg.NRows()))
-	// update timestamp range
+	// update timestamp range and start-end position
 	buffer.updateTimeRange(ibNode.getTimestampRange(tsData))
+	buffer.updateStartAndEndPosition(startPos, endPos)
 
 	metrics.DataNodeConsumeMsgRowsCount.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metrics.InsertLabel).Add(float64(len(msg.RowData)))
 
 	// store in buffer
-	ibNode.insertBuffer.Store(currentSegID, buffer)
-
-	// store current endPositions as Segment->EndPostion
-	ibNode.channel.updateSegmentEndPosition(currentSegID, endPos)
+	ibNode.channel.setCurInsertBuffer(currentSegID, buffer)
```

Comment:
there are two situations of `if !loaded`:
1.  segment not exist，which would not happen, because segment must be created at `updateSegmentStates`;
2. buffer is nil, setCurInsertBuffer would replace the nil buffer, which is expectd.